### PR TITLE
Refactor

### LINKED ISSUE #26: Refactor
这屑代码丑的一匹
让爷给他重构了

一些重构的点：

- [x] 为工具数据结构`WithCase`实现 `Marshaler` 等接口，以便直接从 yaml parse 出来
- [x] 修改yaml结构，使其更容易 parse
- [x] 直接 parse yaml 得到各种数据结构，不再经过各种 compat，即让yamlFile继承/组合自table等
- [x] 更合理的文件夹组织和层次分割

### PR DESCRIPTION
**Pull request by longfangsong**
@zhoudian64  PTAL
This should fix #26

### CODE REVIEW

File: devops/devops.go [hunk 1]
```diff
@@ -1,20 +0,0 @@
-package devops
```
Comment (longfangsong) [inline devops/devops.go:1]: 我觉得 devops 这个词太宽泛了

File: generator/apiGenerator/golang/typeTransformer/typeTransformer.go [hunk 2]
```diff
@@ -1,8 +1,8 @@
-package typeTransform
+package golangTypeTransformer
```
Comment (longfangsong) [inline generator/apiGenerator/golang/typeTransformer/typeTransformer.go:1]: SB golang不支持包名为 golang.TypeTransformer

File: yamlParser/yamlParser.go [hunk 3]
```diff
@@ -1,37 +0,0 @@
-package yamlParser
-
-import (
-	yaml "gopkg.in/yaml.v3"
-	"io/ioutil"
-	"os"
-)
-
-type YamlFile struct {
```
Comment (longfangsong) [inline yamlParser/yamlParser.go:9]: 我思考了一下，这是个config，也不一定要从yaml里面来（只是我们恰好现在选了yaml，以后添加json、toml甚至命令行里来也不是不行），所以`YamlFile`这个名字就不合适

File: configFile/fields/cicd/cicd.go [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package cicd
+
+type CICD struct {
```
Comment (zhoudian64) [inline configFile/fields/cicd/cicd.go:3]: I think pointer is a bad idea. Invalid Pointer is more horrible than string's overhead.

Comment (zhoudian64) [reply configFile/fields/cicd/cicd.go:3]: Feel free to ignore this comment.

File: main.go [hunk 5]
```diff
@@ -30,18 +29,19 @@ func main() {
 		Name:  "autoAPI",
 		Usage: "Generate an CRUD api endpoint program automatically!",
 		Action: func(c *cli.Context) error {
-			f, err := yamlParser.Load(c.String("file"))
+			f, err := configFile.LoadYaml(c.String("file"))
```
Comment (zhoudian64) [inline main.go:32]: We may match LoadMethod by `-f filename.<TYPE>`.

Comment (longfangsong) [reply main.go:32]: > We may match LoadMethod by `-f filename.<TYPE>`.

Maybe in the future.

File: configFile/fields/cicd/cicd.go (hunk 4 above)
Comment (longfangsong) [reply configFile/fields/cicd/cicd.go:3]: > I think pointer is a bad idea. Invalid Pointer is more horrible than string's overhead.

After all, these things will be handled in #33.

**Pull request merged**